jacobian_follower/include/jacobian_follower/jacobian_follower.hpp

```diff
@@ -144,6 +145,17 @@ class JacobianFollower {
   std::vector<std::vector<double>> compute_IK_solutions(geometry_msgs::Pose target_pose,
                                                         const std::string &group_name) const;
 
+  std::optional<moveit_msgs::RobotState> computeCollisionFreePointIK(
```

Comment:
There are a lot of arguments. You might consider defining a wrapper class type to reduce number of arguments.

Downside in this case is you'll need a decoder for pybind.
Upside is less likely to make a mistake when passing arguments across from python

Reply:
I agree there are benefits to a struct, but pybind already gives pretty nice error messages, so for now I'll leave it as is.